### PR TITLE
Hevs/feature get-sosreport script

### PR DESCRIPTION
**Pull request by ehvs**
To assist with troubleshooting.

### CODE REVIEW

**Comment by ehvs** [comment]
I dont expect the ci.ext.devshift to work since changing to double quotes when passing to oc debug/node breaks the function.

**Comment by xiaoyu74** [comment]
@ehvs - As I replied in the [slack discussion](https://redhat-internal.slack.com/archives/C016S65RNG5/p1700078101807549?thread_ts=1699882332.217309&cid=C016S65RNG5), please fix the issue that CI reported below and better to test the script against a staging cluster to see if it's working as expected. 
```
06:26:14 In ./SREP/get-sosreport/script.sh line 53:
06:26:14     oc -n default debug node/"${NODE}" -- bash -c 'cat $(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)' > "${DUMP_DIR}"/sosreport-"${NODE}".tar.xz ;
06:26:14                                                   ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
06:26:14 
06:26:14 make: *** [shellcheck] Error 123
06:26:14 Build step 'Execute shell' marked build as failure
```

**Comment by ehvs** [comment]
Hi @xiaoyu74 as commented here and in the slack thread, using double quotes breaks the command. For that one it needs to be single quote.

**Comment by xiaoyu74** [comment]
@ehvs - I still got the CI build failed for this PR indicates the same single quotes issue. Since our CI integrated the `shellcheck` tool and you have to fix it to make the CI build succeed.  In shell scripting, variables and command substitutions (like `$(...)`) do not get expanded within single quotes `(')`. They only get expanded within double quotes `(")`

May be try to fix like: 
```
oc -n default debug node/"${NODE}" -- bash -c "cat \$(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)" > "${DUMP_DIR}"/sosreport-"${NODE}".tar.xz ;
```

Also, I can see your new created PR included some commits from other people, which messed up this PR commits, you probably created the branch without sync to the upstream first to make it clean.

File: scripts/SREP/get-sosreport/script.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    if ! oc -n default debug node/"${NODE}" -- bash -c "ls -lh /host/var/tmp/sosreport-*.tar.xz" | tee;
+        then printf "========\n [Error] Sosreport file not found in node directory /host/var/tmp dir \n========\n"
+        exit 1
+        else printf "========\n [Success] Sosreport file generated\n========\n"
+    fi
+}
+copy_sosreport() {
+# 2nd Debug session - Fetch sosreport .tar.xz file and save in the container volume.
+    printf "\n==== Copying file from node to container volume ====\n"
+    oc -n default debug node/"${NODE}" -- bash -c 'cat $(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)' > "${DUMP_DIR}"/sosreport-"${NODE}".tar.xz ;
```
Comment (cblecker) [inline scripts/SREP/get-sosreport/script.sh:53]:
```suggestion
    oc -n default debug "node/${NODE}" -- bash -c 'cat $(ls -tA /host/var/tmp/sosreport-*.tar.xz | head -1)' > "${DUMP_DIR}/sosreport-${NODE}.tar.xz" ;
```

You want to double quote things that should result in a single string

**Comment by xiaoyu74** [comment]
Hi @ehvs - are you still working on this PR, anything problem you got or any help needed from us?

**Comment by xiaoyu74** [comment]
Got no response from the PR author in this PR and slack, will close this ticket for now. cc: @bmeng 

https://redhat-internal.slack.com/archives/C016S65RNG5/p1707682362438309

**Pull request closed without merging**